app/services/schools/query.rb

```diff
@@ -0,0 +1,175 @@
+          .where(contract_periods: { year: contract_period_id })
+          .where('schools.id = s.id')
+          .limit(1)
+          .to_sql
+        }
+      ) AS transient_in_partnership"
+    end
+
+    def transient_mentors_at_school(contract_period_id)
+      return School.none if ignore?(filter: contract_period_id) || contract_period_id.blank?
+
+      "EXISTS(
+        #{
+          School.select('1 AS one').from('schools s')
+          .left_joins(mentor_at_school_periods: { training_periods: { expression_of_interest: :contract_period } })
+          .left_joins(mentor_at_school_periods: { training_periods: { active_lead_provider: :contract_period } })
+          .where('contract_periods.year = ? OR contract_periods_active_lead_providers.year = ?', contract_period_id, contract_period_id)
+          .where('schools.id = s.id')
+          .limit(1)
+          .to_sql
+        }
+      ) AS transient_mentors_at_school"
+    end
+
+    def transient_ects_at_school_training_programme(contract_period_id)
+      return School.none if ignore?(filter: contract_period_id) || contract_period_id.blank?
+
+      "(
+        #{
+          School.distinct.select('ect_at_school_periods.training_programme').from('schools s')
+          .left_joins(ect_at_school_periods: { training_periods: { expression_of_interest: :contract_period } })
+          .left_joins(ect_at_school_periods: { training_periods: { active_lead_provider: :contract_period } })
+          .where('contract_periods.year = ? OR contract_periods_active_lead_providers.year = ?', contract_period_id, contract_period_id)
+          .where('schools.id = s.id')
+          .order(training_programme: :asc)
+          .limit(1)
+          .to_sql
+        }
+      ) AS transient_ects_at_school_training_programme"
+    end
+
+    def transient_expression_of_interest(lead_provider_id, contract_period_id)
+      return School.none if ignore?(filter: lead_provider_id) ||
+        lead_provider_id.blank? ||
+        ignore?(filter: contract_period_id) ||
+        contract_period_id.blank?
+
+      mentors_eoi = School.select('1 AS one').from('schools s')
+                  .joins(mentor_at_school_periods: { training_periods: { expression_of_interest: :contract_period } })
+                  .where(contract_periods: { year: contract_period_id })
+                  .where(expression_of_interest: { lead_provider_id: })
+                  .where('schools.id = s.id')
+                  .limit(1)
+                  .to_sql
+
+      ects_eoi = School.select('1 AS one').from('schools s')
+                  .joins(ect_at_school_periods: { training_periods: { expression_of_interest: :contract_period } })
+                  .where(contract_periods: { year: contract_period_id })
+                  .where(expression_of_interest: { lead_provider_id: })
+                  .where('schools.id = s.id')
+                  .limit(1)
+                  .to_sql
+
+      "CASE
```

Comment:
maybe this final logic can be in ruby and we just include the ects/mentors as transient values?